### DISCUSSION TITLE
Understanding project architecture

### DISCUSSION THREAD

**Discussion by ghost** [google/flax#2370] [Q&A] [answered]
Hi guys,

Greetings.

I am learning about python architecture and better practices of how to import packages/modules and I am trying to follow project structures from the community to get the idea. So, my question about flax architecture is related to the `tests` folder. I noticed that each test file that needs to import the module for testing uses `flax` package from `flax/flax` without modifying the `sys.path` with the `__init__.py` file. I was wondering how if this approach for development:

## option 1
1. The flex package is installed locally with an editable setup?

## option 2
2. When running a test file is done with 
 
` PYTHONPATH=path/to/flex python3 test-file.py`

Thanks!

**Comment by levskaya** [comment, marked as answer]
If I understand what you're asking, what we almost always do for development is to use virtualenvs 
(e.g. `mkdir envs; python -m venv my_env; source envs/my_env/bin/activate` then `pip install -e flax_repo_dir/` so that the editable version in the local repo is used by tests during development.
